components/script/dom/globalscope.rs

```diff
@@ -294,15 +294,19 @@ pub struct GlobalScope {
     ///
     /// <https://html.spec.whatwg.org/multipage/#about-to-be-notified-rejected-promises-list>
     #[ignore_malloc_size_of = "mozjs"]
-    uncaught_rejections: DomRefCell<Vec<Heap<*mut JSObject>>>,
+    #[allow(clippy::vec_box)]
+    // `Heap` values must stay boxed, as they need semantics like `Pin` ie they cannot be moved
+    uncaught_rejections: DomRefCell<Vec<Box<Heap<*mut JSObject>>>>,
 
     /// Promises in this list have previously been reported as rejected
     /// (because they were in the above list), but the rejection was handled
     /// in the last turn of the event loop.
     ///
     /// <https://html.spec.whatwg.org/multipage/#outstanding-rejected-promises-weak-set>
     #[ignore_malloc_size_of = "mozjs"]
-    consumed_rejections: DomRefCell<Vec<Heap<*mut JSObject>>>,
+    #[allow(clippy::vec_box)]
+    // `Heap` values must stay boxed, as they need semantics like `Pin` ie they cannot be moved
```

Comment:
I think it makes sense to put this above the `allow` directive here and everywhere.